### PR TITLE
XcodeGen for Student project

### PR DESCRIPTION
**Pull request by szabinst**
[ignore-commit-lint]

## Checklist

- [ ] Follow-up e2e test ticket created
- [ ] A11y checked
- [ ] Tested on phone
- [ ] Tested on tablet
- [ ] Tested in dark mode
- [ ] Tested in light mode
- [ ] Approve from product

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Build failed, skipping coverage check</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 2f202a30cb25062085e20b23937a624d19a895e1
</p>

**Review by vargaat — COMMENTED** [review]
I would leave the file reorganization to another PR as it requires more work since we have a lot of places where resource/source file paths are hardcoded. I checked what was in my mind but there could be other places where we have hardcoded paths, all needs to be modified and tested:

- build-colors.js
- build-icons.js
- build-secrets.js
- translations/export.js
- translations/import.js
- coverage/config.json
- setup-parent-secrets.sh
- setup-student-secrets.sh
- setup-teacher-secrets.sh
- Bitrise workflows to be checked (Info.plist locations, adhoc icon stamping, etc.)

File: Makefile [hunk 1]
```diff
@@ -15,6 +15,7 @@ gen: ## xproj file generation
 	@echo 🟡 Running xcodegen
 	@cd Core; xcodegen
 	@cd TestsFoundation; xcodegen
+#	@cd Student; xcodegen
```
Comment (vargaat) [inline Makefile:18]: Guess this shouldn't be commented out.

File: Student/project.yml [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+        INFOPLIST_FILE: Student/Info.plist
+        CODE_SIGN_ENTITLEMENTS: Student/Student.entitlements
+        TARGETED_DEVICE_FAMILY: 1,2
+    sources:
+      - path: Student
+    dependencies:
+      # - package: GoogleUtilities
+
+      # - package: Firebase
+      #   product: FirebaseRemoteConfig
```
Comment (vargaat) [inline Student/project.yml:57]: What are these for? Let's add some explanation or remove them.

File: Student/project.yml [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    settings:
+      APPLICATION_EXTENSION_API_ONLY: true
+      ENABLE_TESTABILITY: true
+      base:
+        PRODUCT_BUNDLE_IDENTIFIER: com.instructure.icanvas.SubmitAssignment
+        INFOPLIST_FILE: SubmitAssignment/Info.plist
+        CODE_SIGN_ENTITLEMENTS: SubmitAssignment/SubmitAssignment.entitlements
+        TARGETED_DEVICE_FAMILY: 1,2
+    dependencies:
+      # - package: Firebase
```
Comment (vargaat) [inline Student/project.yml:85]: Can we remove this?

File: Student/project.yml [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+    platform: iOS
+    settings:
+      APPLICATION_EXTENSION_API_ONLY: false
+      ENABLE_TESTABILITY: true
+      TEST_HOST: $(BUILT_PRODUCTS_DIR)/Student.app/Student
+    sources:
+      - path: StudentUnitTests
+    #dependencies:
+    #  - target: Core/Core
+    #  - target: TestsFoundation/TestsFoundation
```
Comment (vargaat) [inline Student/project.yml:116]: Let's remove these if not needed.

**Review by vargaat — CHANGES_REQUESTED** [review]
Let's finalize the PR and run every CI workflow on this branch to be sure nothing is broken.

File: Makefile (hunk 1 above)
Comment (szabinst) [reply Makefile]: Readded this line.

File: Canvas.xcworkspace/xcshareddata/swiftpm/Package.resolved
Comment (vargaat) [inline Canvas.xcworkspace/xcshareddata/swiftpm/Package.resolved:1]: Were these version downgrades intentional?

**Review by vargaat — CHANGES_REQUESTED** [review]

**Review by vargaat — APPROVED** [review]
QA+1, cocoapods mixes up a few things but hopefully not for long

**Pull request merged**